File: Stack/Basic Calculator/BasicCalculator.cs
public class Solution
{
    public int Calculate(string s)
    {
        int output = 0, curr = 0, sign = 1;
        Stack<int> stack = new Stack<int>();
        for (int i = 0; i < s.Length; i++)
        {
            char c = s[i];
            if (char.IsDigit(c))
            {
                curr = (curr * 10) + (c - '0');
            }
            else if (c == '+' || c == '-')
            {
                output += curr * sign;
                curr = 0;
                sign = (c == '+') ? 1 : -1;
            }
            else if (c == '(')
            {
                stack.Push(output);
                stack.Push(sign);
                output = 0;
                sign = 1;
            }
            else if (c == ')')
            {
                output += curr * sign;
                output *= stack.Pop();
                output += stack.Pop();
                curr = 0;
            }
        }
        return output + (curr * sign);
    }
}

File: Stack/Basic Calculator/BasicCalculator.py
class Solution:
    def calculate(self, s: str) -> int:
        output, curr, sign, stack = 0, 0, 1, []
        for i in range(len(s)):
            c = s[i]
            if c.isdigit():
                curr = (curr * 10) + int(c)
            elif c in "+-":
                output += curr * sign
                curr = 0
                sign = 1 if c == "+" else -1
            elif c == "(":
                stack.append(output)
                stack.append(sign)
                output = 0
                sign = 1
            elif c == ")":
                output += curr * sign
                output *= stack.pop()
                output += stack.pop()
                curr = 0
        return output + (curr * sign)
